### PR TITLE
Go func ref annotations

### PR DESCRIPTION
**Pull request by gpene**
@seanpackham for whoever we get to review this, the following need the more careful review/are missing definitions where I was unsure:
* MetricsCounterAdd
* MetricsGaugeSet
* MetricsTimerRecord
* SetEventFn
* ChannelIdBuild

### CODE REVIEW

File: server/runtime_go_nakama.go [hunk 1]
```diff
@@ -2524,6 +3268,12 @@ func (n *RuntimeGoNakamaModule) ChannelMessageUpdate(ctx context.Context, channe
 	return ChannelMessageUpdate(ctx, n.logger, n.db, n.router, channelIdToStreamResult.Stream, channelId, messageId, contentStr, senderId, senderUsername, persist)
 }
 
+// @summary Create a chat channel.
```
Comment (mofirouz) [inline server/runtime_go_nakama.go:3271]: Helper function to create an channel identifier that can be used for various other runtime calls; Note, this does NOT create a channel.

Target: Can be the room name, group identifier, or another username

channelId: The generated ID that represents a channel.

File: server/runtime_go_nakama.go [hunk 2]
```diff
@@ -2488,6 +3213,15 @@ func (n *RuntimeGoNakamaModule) SetEventFn(fn RuntimeEventCustomFunction) {
 	n.Unlock()
 }
 
+// @summary Send a message on a realtime chat channel.
```
Comment (mofirouz) [inline server/runtime_go_nakama.go:3216]: SetEventFn - Register a function that processes events published to the server.

File: server/runtime_go_nakama.go [hunk 3]
```diff
@@ -2443,18 +3147,39 @@ func (n *RuntimeGoNakamaModule) Event(ctx context.Context, evt *api.Event) error
...
+// @param tags(map[string]string) The metrics tags associated with this gauge.
+// @param value(float64) 
 func (n *RuntimeGoNakamaModule) MetricsGaugeSet(name string, tags map[string]string, value float64) {
 	n.metrics.CustomGauge(name, tags, value)
 }
 
+// @summary Add a custom metrics timer.
+// @param name(string) The name of the custom metrics timer.
+// @param tags(map[string]string) The metrics tags associated with this timer.
+// @param value(time.Duration)
```
Comment (mofirouz) [inline server/runtime_go_nakama.go:3169]: Value - Value to update this metrics with.

File: server/runtime_go_nakama.go [hunk 4]
```diff
@@ -2443,18 +3147,39 @@ func (n *RuntimeGoNakamaModule) Event(ctx context.Context, evt *api.Event) error
...
+// @param tags(map[string]string) The metrics tags associated with this counter.
+// @param delta(int64) 
 func (n *RuntimeGoNakamaModule) MetricsCounterAdd(name string, tags map[string]string, delta int64) {
 	n.metrics.CustomCounter(name, tags, delta)
 }
 
+// @summary Add a custom metrics gauge.
+// @param name(string) The name of the custom metrics gauge.
+// @param tags(map[string]string) The metrics tags associated with this gauge.
+// @param value(float64)
```
Comment (mofirouz) [inline server/runtime_go_nakama.go:3161]: Value - Value to update this metrics with.

File: server/runtime_go_nakama.go [hunk 5]
```diff
@@ -2443,18 +3147,39 @@ func (n *RuntimeGoNakamaModule) Event(ctx context.Context, evt *api.Event) error
 	return nil
 }
 
+// @summary Add a custom metrics counter.
+// @param name(string) The name of the custom metrics counter.
+// @param tags(map[string]string) The metrics tags associated with this counter.
+// @param delta(int64)
```
Comment (mofirouz) [inline server/runtime_go_nakama.go:3153]: Delta - Value to update this metrics with.

**Review by mofirouz — CHANGES_REQUESTED** [review]
Absolutely fantastic work; Left some comments as you mentioned.

**Comment by gpene** [comment]
Thanks @mofirouz - updated with your comments.

**Review by mofirouz — APPROVED** [review]
Perfect, well done 👍

**Review by zyro — APPROVED** [review]

**Pull request merged**